File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Expressions/Literals/StringLiteral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.QL.AST.Model.Statements;
using UvA.SoftCon.Questionnaire.Common.AST;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions.Literals
{
    /// <summary>
    /// Represents a static, immutable string value.
    /// </summary>
    public class StringLiteral : Literal<string>
    {
        public override bool IsValid
        {
            get
            {
                return true;
            }
        }

        internal StringLiteral(string value, TextPosition position)
            : base(value, position) { }

        public override T Accept<T>(IQuestionFormVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override DataType GetType(IDictionary<string, DataType> symbolTable)
        {
            return DataType.String;
        }

        public override string GetValue()
        {
            return Value;
        }

        public override string ToString()
        {
            return '"' + Value + '"';
        }
    }
}

File: encoders/ql/src/main/java/nl/uva/sc/encoders/ql/ast/expression/literal/StringLiteral.java
package nl.uva.sc.encoders.ql.ast.expression.literal;

import nl.uva.sc.encoders.ql.ast.type.DataType;
import nl.uva.sc.encoders.ql.ast.type.StringType;
import nl.uva.sc.encoders.ql.visitor.LiteralVisitor;

public class StringLiteral implements Literal {

	private final String value;

	public StringLiteral(String value) {
		this.value = value;
	}

	public String getValue() {
		return value;
	}

	@Override
	public DataType getType() {
		return new StringType();
	}

	@Override
	public <T> T accept(LiteralVisitor<T> visitor) {
		return visitor.visit(this);
	}
}
